orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/UpdateManifestWaitForStableTask.java

```diff
@@ -0,0 +1,104 @@
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.manifest;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.OverridableTimeoutRetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.clouddriver.model.Manifest;
+import com.netflix.spinnaker.orca.clouddriver.utils.CloudProviderAware;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@Component
+@Slf4j
+public class UpdateManifestWaitForStableTask implements OverridableTimeoutRetryableTask, CloudProviderAware {
+  public final static String TASK_NAME = "waitForStable";
+
+  @Autowired
+  OortService oortService;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  @Override
+  public long getBackoffPeriod() {
+    return TimeUnit.SECONDS.toMillis(1);
+  }
+
+  @Override
+  public long getTimeout() {
+    return TimeUnit.MINUTES.toMillis(30);
+  }
+
+
+  @Override
+  public TaskResult execute(Stage stage) {
+    String account = getCredentials(stage);
+    String name = (String) stage.getContext().get("manifest.name");
+    String location = (String) stage.getContext().get("manifest.location");
+    String id = String.format("'%s' in '%s' for account %s", name, location, account);
+    List<String> messages = new ArrayList<>();
+
+    Response response;
+    try {
+      response = oortService.getManifest(account, location, name);
+    } catch (RetrofitError e) {
+      log.warn("Unable to read manifest {}", id, e);
+      return new TaskResult(ExecutionStatus.RUNNING, new HashMap<>(), new HashMap<>());
+    }
+
+    if (response.getStatus() != 200) {
+      return new TaskResult(ExecutionStatus.RUNNING, new HashMap<>(), new HashMap<>());
+    }
+
+    Manifest manifest;
+    try {
+      manifest = objectMapper.readValue(response.getBody().in(), Manifest.class);
+    } catch (IOException e) {
+      log.error("Failed to read " + id, e);
+      throw new RuntimeException(e);
+    }
+
+    Manifest.Status status = manifest.getStatus();
+    if (status.getStable() == null || !status.getStable().isState()) {
+      messages.add(status.getStable().getMessage());
+      Map<String, Object> context = new ImmutableMap.Builder<String, Object>()
+          .put("stableMessages", messages)
+          .build();
+
+      return new TaskResult(ExecutionStatus.RUNNING, context, new HashMap<>());
```

Comment:
This is going to overwrite any `stableMessages` that were added previously, is that likely to be an issue in reality?

Reply:
That won't be a problem - however - 

I thought that with the new non-global context that @robfletcher has been working on migrating to we could look up any stage's outputs, and even aggregate across all stages that emit a certain key?

Reply:
I think he might be asking about the sequence of `stableMessages` generated by retries of this task. Which might be interesting.

Reply:
Ahh, that's definitely worth overwriting